src/utility/stm32_eth.c

```diff
@@ -45,6 +45,7 @@
 #include "lwip/dhcp.h"
 #include "lwip/prot/dhcp.h"
 #include "lwip/dns.h"
+#include "core_callback.h"
```

Comment:
To avoid any build issue if user do not use the next core version and ensure compatibility, it should be fine to do something like that? Tested and it's ok.
```
#if __has_include("core_callback.h")
#include "core_callback.h"
#else
void registerCoreCallback(void (*func)(void)) {
  UNUSED(func);
}
#endif
```